addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1395 @@
+
+							GetIdealMovementsInAir(angles[1], fVelocity2D, fMaxSpeed, fSurfaceFriction, fFowardMove, fSideMove);
+
+							float fAngleDifference = AngleNormalize(angles[1] - gF_LastAngle[client]);
+							float fCurrentAngles = FloatAbs(fAngleDifference);
+
+
+							// Right
+							if (fAngleDifference < 0.0)
+							{
+								float fMaxDelta = GetMaxDeltaInAir(fVelocity2D, fMaxSpeed, fSurfaceFriction, true);
+								vel[1] = gF_MaxMove;
+
+								if (fCurrentAngles <= fMaxDelta * gF_Power[client])
+								{
+									vel[0] = fFowardMove * gF_MaxMove;
+									vel[1] = fSideMove * gF_MaxMove;
+								}
+							}
+							else if (fAngleDifference > 0.0)
+							{
+								float fMaxDelta = GetMaxDeltaInAir(fVelocity2D, fMaxSpeed, fSurfaceFriction, false);
+								vel[1] = -gF_MaxMove;
+
+								if (fCurrentAngles <= fMaxDelta * gF_Power[client])
+								{
+									vel[0] = fFowardMove * gF_MaxMove;
+									vel[1] = fSideMove * gF_MaxMove;
+								}
+							}
+							else
+							{
+								vel[0] = fFowardMove * gF_MaxMove;
+								vel[1] = fSideMove * gF_MaxMove;
+							}
+						}
+					}
+				}
+				/*
+							WIGGLEHACK END
+														*/
+
+				if(gF_TimescaleTicksPassed[client] >= 1.0)
+				{
+					gF_TimescaleTicksPassed[client] -= 1.0;
+
+					gF_TASTime[client] += GetTickInterval();
+
+					int iFrameNumber = GetArraySize(gA_Frames[client])+1;
+					if(gI_IndexCounter[client] != iFrameNumber-2)
+					{
+						//UnPaused in different tick
+						iFrameNumber = gI_IndexCounter[client]+1;
+					}
+					ResizeArray(gA_Frames[client], iFrameNumber);
+					
+					float fPosition[3];
+					float fEyeAngles[3];
+					float fVelocity[3];
+
+					GetEntPropVector(client, Prop_Send, "m_vecOrigin", fPosition);
+					GetClientEyeAngles(client, fEyeAngles);
+					GetEntPropVector(client, Prop_Data, "m_vecAbsVelocity", fVelocity);
+
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fPosition[0], 0);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fPosition[1], 1);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fPosition[2], 2);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fEyeAngles[0], 3);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fEyeAngles[1], 4);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, buttons, 5);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, GetEntityFlags(client), 6);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, GetEntityMoveType(client), 7);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fVelocity[0], 8);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fVelocity[1], 9);
+					SetArrayCell(gA_Frames[client], iFrameNumber-1, fVelocity[2], 10);
+					gI_IndexCounter[client] = iFrameNumber-1;
+					gF_IndexCounter[client] = iFrameNumber-1.0;
+				}
+				else if(!(GetEntityFlags(client) & FL_ONGROUND))
+				{
+					vel[0] = 0.0;
+					vel[1] = 0.0;
+				}
+
+				// Fix boosters
```

Comment:
I have this sneaky fealing you ripped this from somewhere and that there should be a credit to the original.